### ISSUE TITLE
Should we use Twisted for system calls and asynchronous reads?

### ISSUE THREAD

**Issue by leeping** [leeping/forcebalance#50]
Currently ForceBalance has a method called `_exec()` that it uses for all system calls.  It is a wrapper around functions in `subprocess` that actually carry out the system call, but it uses `select` to poll for whether the stdout and stderr file descriptors have data to read.  This way, we are able to read from both streams while they are being generated without them being garbled.

I have noticed that when using this method in other contexts, sometimes I get the error `file descriptor out of range` or something similar.  This is due to a limitation in `select` where there is a maximum number of file descriptors.  One possibility is that I simply need to close the file descriptor after I'm done reading from it, but I haven't tested this (the error doesn't show up often since it needs to hit the limit).  That would be a simple fix.

Another option would be to bundle Twisted, an event-driven programming library (http://twistedmatrix.com), with ForceBalance.  It abstracts away low-level things like `select` and should provide the functionality we need, and I think it properly accounts for the file descriptor issue.  It is licensed with the MIT license (highly permissive) and its only dependency is `zope.interface`.  Having better asynchronous abilities in ForceBalance might enable other things as well, including maybe a way to fork local targets so they may be evaluated in parallel.  However, introducing new dependencies should be thought out carefully.

Let me know what you think.  This is medium to low priority.
